modules/indexing/src/main/scala/ch/epfl/bluebrain/nexus/service/indexer/cache/KeyValueStoreSubscriber.scala

```diff
@@ -0,0 +1,111 @@
+package ch.epfl.bluebrain.nexus.service.indexer.cache
+
+import akka.actor.{Actor, ActorRef, ActorSystem, Props}
+import akka.cluster.ddata.LWWMapKey
+import akka.cluster.ddata.Replicator.Changed
+import ch.epfl.bluebrain.nexus.service.indexer.cache.KeyValueStoreSubscriber.KeyValeStoreChange._
+import ch.epfl.bluebrain.nexus.service.indexer.cache.KeyValueStoreSubscriber._
+
+trait OnKeyValueStoreChange[K, V] {
+
+  /**
+    * Method that gets triggered when a change to key value store occurs.
+    *
+    * @param value the changes made
+    */
+  def apply(value: KeyValeStoreChanges[K, V]): Unit
```

Comment:
I'm wondering if we should make this take an `F[_]: Async` as most of the time I suspect we'd want to do something with the changes and our API is almost all defined like this... thoughts?